src/app/qgsmaptoolreshape.cpp

```diff
@@ -100,6 +100,54 @@ void QgsMapToolReshape::cadCanvasReleaseEvent( QgsMapMouseEvent * e )
         reshapeReturn = geom->reshapeGeometry( points() );
         if ( reshapeReturn == 0 )
         {
+          //avoid intersections on polygon layers
+          if (vlayer->geometryType() == QGis::Polygon) {
+
+            //ignore all current layer features as they should be reshaped too
+            QMap<QgsVectorLayer*, QSet<QgsFeatureId>> ignoreFeatures;
+            ignoreFeatures.insert(vlayer, vlayer->allFeatureIds());
+
+            int avoidIntersectionsReturn = geom->avoidIntersections(ignoreFeatures);
+            if ( avoidIntersectionsReturn == 1 )
+            {
+              //not a polygon type. Impossible to get there
+            }
+#if 0
```

Comment:
Is this code copied from somewhere? Based on the comment, this line was written by @mhugent.

If yes, I wonder if it could be refactored into a new method to maintain this code only once.